### ISSUE TITLE
turns should not be possible while using blowtorch

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#249]
```
What steps will reproduce the problem?
 * see the video at http://www.youtube.com/watch?v=a9-3_8NQpUk

What is the expected output? What do you see instead?
 * it should not be possible for user to turn 180 degrees once blowtorch is activated

What version of the product are you using? On what operating system?
 * 0.9.15 Ubuntu 10.04


```

-----
Original issue reported on code.google.com by `jose1...@gmail.com` on 19 Jul 2011 at 11:58

**Comment by GoogleCodeExporter** [comment]
```
Thank you for reporting.

I wasn't able to reproduce the behavior/bug described above.

Please provide more details :)
```

-----
Original comment by `sheepyluva` on 19 Jul 2011 at 12:42

**Comment by GoogleCodeExporter** [comment]
```
Oh... and please upload the demo of that game to the website if you don't know 
how to do it yourself and if you were just watching the other guy do it.
```

-----
Original comment by `sheepyluva` on 19 Jul 2011 at 12:44

**Comment by GoogleCodeExporter** [comment]
```
i was only watching it. but the guy who did that said he was only aiming 
up/down when to his surprise it happened. link to demo: 
http://dl.dropbox.com/u/29095940/blowtorch-bug.hwd
```

-----
Original comment by `jose1...@gmail.com` on 19 Jul 2011 at 12:52

**Comment by GoogleCodeExporter** [comment]
```
Ah ok thanks. I think I know what happened.

The kicked hedgehog bounced back and kicked/turned the blowtorch hedgehog into 
the other direction.

Interesting find, not sure if that is considered a bug though.
I'll ask the game designer/devs.

Thanks for the upload!
```

-----
Original comment by `sheepyluva` on 19 Jul 2011 at 1:04

**Comment by GoogleCodeExporter** [comment]
```
Yeah, as we were discussing in IRC, I've known about it for a while since I 
play Clean Slate a lot, and it is pretty obvious what happens after you've run 
into it a couple of times.

I just hadn't jumped on it since, well, it was pretty rare event and could 
usually be avoided with just a little practice anyway.

Workaround could be to keep track of dX of hog, and if X direction changes 
while still being a low enough value (since we still want to allow hogs to be 
kicked by mines/barrels), restore the old direction/velocity.

```

-----
Original comment by `kyberneticist@gmail.com` on 19 Jul 2011 at 4:33

**Comment by GoogleCodeExporter** [comment]
-----
Original comment by `sheepyluva` on 19 Jul 2011 at 6:01
* Changed state: **Accepted**
* Added labels: **Priority-Low**
* Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
This behaviour is fixed in revision 3b7f2bfc8632, even though hog itself still 
could  turn around, blowtorch won't change its direction and will turn hedgehog 
to proper direction anyway
```

-----
Original comment by `unC0Rr` on 2 Sep 2011 at 7:17
* Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
verified (@ low velocities, hog w/ torch can't turn)
```

-----
Original comment by `jerryma1...@gmail.com` on 21 Dec 2011 at 12:42
* Changed state: **Verified**
